langchain/src/util/tiktoken.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Tiktoken,
+  TiktokenBPE,
+  TiktokenEncoding,
+  TiktokenModel,
+  getEncodingNameForModel,
+} from "js-tiktoken/lite";
+
+const cache: Record<string, TiktokenBPE> = {};
+
+export async function getEncoding(
```

Comment:
Is it possible this will ever be called multiple times concurrently? If so (and not yet cached) we'd end up with multiple inflight requests for the same thing. If we think that's likely can avoid by also caching the promise synchronously